### DISCUSSION TITLE
Add to keep an element in the DOM when element in focus, or other case

### DISCUSSION THREAD

**Discussion by pan-alexey** [TanStack/virtual#878] [Ideas]
Hi.

We are implementing a table with a virtual list and have encountered a problem with keyboard navigation. During scrolling, the element that is currently in focus can be removed from the DOM, causing us to lose focus.

Is there a way to specify that this focused element should not be removed from the DOM tree, so that the browser does not lose focus?

**Comment by piecyk** [comment]
Yes, you can use `rangeExtractor` to keep the focus index always mounted https://tanstack.com/virtual/v3/docs/api/virtualizer#rangeextractor
